libsrc/leddevice/dev_hid/LedDeviceMultiLightpack.h

```diff
@@ -19,47 +20,73 @@
 class LedDeviceMultiLightpack : public LedDevice
 {
 public:
+
+	///
+	/// @brief Constructs a LedDevice of multiple Lightpack LED-devices
 	///
-	/// Constructs specific LedDevice
+	/// @param deviceConfig Device's configuration as JSON-Object
 	///
-	explicit LedDeviceMultiLightpack(const QJsonObject &);
+	explicit LedDeviceMultiLightpack(const QJsonObject &deviceConfig);
 
 	///
-	/// Destructor of the LedDevice; closes the output device if it is open
+	/// @brief Destructor of the LedDevice
 	///
 	virtual ~LedDeviceMultiLightpack() override;
 
-	/// constructs leddevice
+	///
+	/// @brief Constructs the LED-device
+	///
+	/// @param[in] deviceConfig Device's configuration as JSON-Object
+	/// @return LedDevice constructed
+	///
 	static LedDevice* construct(const QJsonObject &deviceConfig);
 
+protected:
+
+	///
+	/// @brief Initialise the device's configuration
 	///
-	virtual int switchOff() override;
+	/// @param[in] deviceConfig the JSON device configuration
+	/// @return True, if success
+	///
+	virtual bool init(const QJsonObject &deviceConfig) override;
 
-protected:
 	///
-	/// Opens and configures the output device7
+	/// @brief Opens the output device.
 	///
-	/// @return Zero on succes else negative
+	/// @return Zero on success (i.e. device is ready), else negative
 	///
-	int open() override;
+	virtual int open() override;
+
+	///
+	/// @brief Closes the output device.
 	///
-	/// Switch the leds off
+	/// @return Zero on success (i.e. device is closed), else negative
 	///
-	/// @return Zero on success else negative
+	virtual int close() override;
 
-private:
 	///
-	/// Writes the RGB-Color values to the leds.
+	/// @brief Power-/turn off the Nanoleaf device.
 	///
-	/// @param[in] ledValues  The RGB-color per led
+	/// @return True if success
+	///
+	virtual bool powerOff() override;
+
 	///
-	/// @return Zero on success else negative
+	/// @brief Writes the RGB-Color values to the LEDs.
 	///
-	virtual int write(const std::vector<ColorRgb>& ledValues) override;
+	/// @param[in] ledValues The RGB-color per LED
+	/// @return Zero on success, else negative
+	///
+	virtual int write(const std::vector<ColorRgb> & ledValues) override;
```

Comment:
If virtual or not is there does not make any difference, or ? It may just outline the function is virtual from a documentation perspective....

_"If some member function vf is declared as virtual in a class Base, and some class Derived, which is derived, directly or indirectly, from Base, has a declaration for member function with the same
    name
    parameter type list (but not the return type)
    cv-qualifiers
    ref-qualifiers 
**Then this function in the class Derived is also virtual (whether or not the keyword virtual is used in its declaration)** and overrides Base::vf (whether or not the word override is used in its declaration). "_

Reply:
It doesn't make any difference, just a matter of preference.

https://clang.llvm.org/extra/clang-tidy/checks/modernize-use-override.html

Go as you wish :)

Reply:
You have a point. But I am hesitant to update all header files again :-(
Maybe there is an auto-cleanup function....

Reply:
Yeah those should be done via clang-tidy. No need for manual cleanup. These can be fixed all at once when we setup a proper clangtidy file across the project.